cfour/default.nix

```diff
@@ -0,0 +1,66 @@
+{ stdenv, lib, makeWrapper, gfortran, requireFile, writeTextFile, python3, blas-i8, lapack,
+  mpi ? null
+}:
+# Make sure to have a correct version of BLAS.
+assert
+  lib.asserts.assertMsg
+  (blas-i8.isILP64)
+  "A 64 bit integer implementation of BLAS is required.";
+
+# Only works with OpenMPI
+assert
+  lib.asserts.assertMsg
+  (mpi == null || mpi.pname == "openmpi")
+  "If MPI is to be used for CFour, use OpenMPI!";
+let
+  useMPI = mpi != null;
```

Comment:
Is there a legitimate use-case for using it without MPI? Would it make sense to remove the conditional and just build it always with MPI?

Reply:
The OMP parallel version is also working nice. We could always build with MPI, but i haven't solved the problem mentioned in the original PR message with the number of cores yet.

Reply:
OK, let's keep the conditional.